packages/material-ui/src/Slide/Slide.js

```diff
@@ -161,84 +146,100 @@ class Slide extends React.Component {
       ...transitionProps,
       easing: theme.transitions.easing.sharp,
     });
-    setTranslateValue(this.props, node);
+    setTranslateValue(direction, node);
 
-    if (this.props.onExit) {
-      this.props.onExit(node);
+    if (onExit) {
+      onExit(node);
     }
   };
 
-  handleExited = node => {
+  const handleExited = () => {
+    const node = childrenRef.current;
     // No need for transitions when the component is hidden
     node.style.webkitTransition = '';
     node.style.transition = '';
 
-    if (this.props.onExited) {
-      this.props.onExited(node);
+    if (onExited) {
+      onExited(node);
     }
   };
 
-  /**
-   * used in cloneElement(children, { ref: handleRef })
-   */
-  handleRef = ref => {
-    // #StrictMode ready
-    this.childDOMNode = ReactDOM.findDOMNode(ref);
-    setRef(this.props.children.ref, ref);
-  };
+  const updatePosition = React.useCallback(() => {
+    if (childrenRef.current) {
+      setTranslateValue(direction, childrenRef.current);
+    }
+  }, [direction]);
+
+  React.useEffect(() => {
+    const handleResize = debounce(() => {
+      // Skip configuration where the position is screen size invariant.
+      if (inProp || direction === 'down' || direction === 'right') {
+        return;
+      }
+
+      if (childrenRef.current) {
+        setTranslateValue(direction, childrenRef.current);
+      }
+    }, 166); // Corresponds to 10 frames at 60 Hz.
 
-  updatePosition() {
-    if (this.childDOMNode) {
-      setTranslateValue(this.props, this.childDOMNode);
+    window.addEventListener('resize', handleResize);
+
+    return () => {
+      handleResize.clear();
+      window.removeEventListener('resize', handleResize);
+    };
+  }, [direction, inProp]);
+
+  React.useEffect(() => {
+    // state.mounted handle SSR, once the component is mounted, we need
+    // to properly hide it.
+    if (!inProp && !mountedRef.current) {
```

Comment:
Isn't this just a second call to `updatePosition` now on the first effect? Looks like we can get rid of this call in favor of https://github.com/mui-org/material-ui/pull/15344/files#diff-bfee2b0644a8d8af8757fa2331fb8e65R204

Edit: Fairly certain this is just a remnant of cDM. `useEffect` takes care of both lifecycles :)